src/Hl7.Fhir.Core/Serialization/ComplexTypeWriter.cs

```diff
@@ -47,120 +49,104 @@ internal void Serialize(ClassMapping mapping, object instance, Rest.SummaryType
 
             _writer.WriteStartComplexContent();
 
-            // Emit members that need xml /attributes/ first (to facilitate stream writer API)
-            foreach (var prop in mapping.PropertyMappings.Where(pm => pm.SerializationHint == XmlSerializationHint.Attribute))
+            // Emit members that need xml attributes / first (to facilitate stream writer API)
+            // attributes first (xml) and order maintained for the rest
+            var propertiesToWrite = mapping.PropertyMappings
```

Comment:
The original code *always* wrote the "id" property but I think the net result is the same: id is marked as "in summary" (for 'true'), explicitly included (for 'text' and 'count'), not excluded (for 'data'), or included (for 'false').